src/formatDate.js

```diff
@@ -51,6 +51,34 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  const dateArray = date.split(fromFormat[3]);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObject[fromFormat[i]] = dateArray[i];
+  }
+
+  for (let i = 0; i < 3; i++) {
+    const oldFormat = fromFormat[i];
+    const index = toFormat.indexOf(oldFormat);
+
+    if (index === -1) {
+      if (oldFormat === 'YY') {
+        dateObject['YYYY'] = Number(dateArray[i]) < 30 
+          ? 20 + dateArray[i] 
+          : 19 + dateArray[i];
+      }
+
+      if (oldFormat === 'YYYY') {
+        dateObject['YY'] = dateArray[i].slice(2);
+      }
+    }
+
+    newDate += dateObject[toFormat[i]] + toFormat[3];
+  }
+
+  return newDate.slice(0, -1);
```

Comment:
why do we need to add `toFormat[3]` each times?
you can define newDate like array, push formatted element, and return joined array with separator

Reply:
Thanks, done!